build.gradle

```diff
@@ -1,14 +1,224 @@
+        })
+    }
+
+    ext {
+        spineProtobufPluginId = 'io.spine.tools.spine-model-compiler'
+
+        sourcesRootDir = "$projectDir/src"
+        generatedRootDir = "$projectDir/generated"
+
+        generatedJavaDir = "$generatedRootDir/main/java"
+        generatedTestJavaDir = "$generatedRootDir/test/java"
+
+        generatedGrpcDir = "$generatedRootDir/main/grpc"
+        generatedTestGrpcDir = "$generatedRootDir/test/grpc"
+
+        generatedSpineDir = "$generatedRootDir/main/spine"
+        generatedTestSpineDir = "$generatedRootDir/test/spine"
+    }
+
+    apply plugin: 'java-library'
+    apply plugin: 'com.google.protobuf'
+    apply plugin: 'net.ltgt.errorprone'
+    apply plugin: 'maven-publish'
+
+    sourceCompatibility = 1.8
+    targetCompatibility = 1.8
+
+    repositories {
+        jcenter()
+
+        google()
+        mavenCentral()
+        mavenLocal()
+        maven { url = repos.gradlePlugins }
+        maven { url = repos.spine }
+        maven { url = repos.spineSnapshots }
+    }
+
+    dependencies {
+        implementation deps.build.guava
+        implementation deps.build.slf4j
+        implementation deps.build.checkerAnnotations
+        implementation deps.build.errorProneAnnotations
+
+        testImplementation deps.test.slf4j
+        testImplementation deps.test.guavaTestlib
+        testImplementation deps.test.junit5Api
+        testImplementation deps.test.junit5Runner
+        testImplementation deps.test.mockito
+        testImplementation deps.test.hamcrest
+        testImplementation "io.spine:spine-testutil-client:$spineVersion"
+    }
+
+    sourceSets {
+        main {
+            java.srcDirs generatedJavaDir, "$sourcesRootDir/main/java", generatedSpineDir
+            resources.srcDirs "$sourcesRootDir/main/resources", "$generatedRootDir/main/resources"
+        }
+        test {
+            java.srcDirs generatedTestJavaDir, "$sourcesRootDir/test/java", generatedTestSpineDir
+            resources.srcDirs "$sourcesRootDir/test/resources", "$generatedRootDir/test/resources"
+        }
+    }
+
+    tasks.withType(JavaCompile) {
+        // Exclude generated sources from being analyzed by Error Prone.
+        options.compilerArgs += [
+                '-XepExcludedPaths:.*/generated/.*'
+        ]
+    }
+
+    task sourceJar(type: Jar) {
+        from sourceSets.main.allJava
+        classifier "sources"
+    }
+
+    task testOutputJar(type: Jar) {
+        from sourceSets.test.output
+        classifier "test"
+    }
+
+    task javadocJar(type: Jar, dependsOn: 'javadoc') {
+        from "$projectDir/build/docs/javadoc"
+        classifier "javadoc"
+    }
+
+    test {
+        useJUnitPlatform {
+            includeEngines 'junit-jupiter'
+        }
+        testLogging.showStandardStreams = true
+        include 'io/spine/**'
+
+        afterSuite { final desc, final result ->
+            if (!desc.parent) {
+                logger.lifecycle(
+"""
+Test summary:
```

Comment:
This section is strange. It requires some comments + proper formatting.

BTW, we need to fail the build in case there is at least one failed test.